### PR TITLE
[BR]supress gc when scheduled backup is newly created

### PR DESCRIPTION
**Pull request by olivia-chen-github**
### What problem does this PR solve?

Issue for scheduled backup https://github.com/pingcap/tidb-operator/issues/4939
### What is changed and how does it work?

suppress gr when a scheduled backup is newly created.  
### Code changes

- [x] Has Go code change
- [ ] Has CI related scripts change

### Tests

- [x] Unit test 
- [x] E2E test 
- [ ] Manual test 
- [ ] No code 

### Side effects

- [ ] Breaking backward compatibility 
- [ ] Other side effects: 

### Related changes

- [ ] Need to cherry-pick to the release branch 
- [ ] Need to update the documentation 

### Release Notes

Please refer to [Release Notes Language Style Guide](https://github.com/pingcap/tidb-operator/blob/master/docs/release-note-guide.md) before writing the release note.

```release-note

```

### CODE REVIEW

**Comment by csuzhangxc** [comment]
@grovecai PTAL

File: pkg/backup/backupschedule/backup_schedule_manager.go [hunk 1]
```diff
@@ -542,6 +546,10 @@ func calExpiredBackupsWithLogBackupOn(backupsList []*v1alpha1.Backup, expiredTSO
 	)
 
 	for ; i < len(backupsList); i++ {
+		// the backup status CommitTs will be empty after created. without this, all newly created backups will be GC'ed
+		if len(backupsList[i].Status.CommitTs) == 0 && (v1alpha1.IsBackupScheduled(backupsList[i]) || v1alpha1.IsBackupRunning(backupsList[i]) || v1alpha1.IsBackupPrepared(backupsList[i])) {
```
Comment (grovecai) [inline pkg/backup/backupschedule/backup_schedule_manager.go:550]: We can do this at line-358 instead.

File: pkg/backup/backupschedule/backup_schedule_manager.go [hunk 2]
```diff
@@ -479,6 +479,10 @@ func caculateExpiredBackups(backupsList []*v1alpha1.Backup, reservedTime time.Du
 	expiredTS := config.TSToTSO(time.Now().Add(-1 * reservedTime).Unix())
 	i := 0
 	for ; i < len(backupsList); i++ {
+		// the backup status CommitTs will be empty after created. without this, all newly created backups will be GC'ed
+		if len(backupsList[i].Status.CommitTs) == 0 && (v1alpha1.IsBackupScheduled(backupsList[i]) || v1alpha1.IsBackupRunning(backupsList[i]) || v1alpha1.IsBackupPrepared(backupsList[i])) {
```
Comment (grovecai) [inline pkg/backup/backupschedule/backup_schedule_manager.go:483]: We filter out newly created backups at line-358 instead.

File: pkg/backup/backupschedule/backup_schedule_manager.go (hunk 1 above)
Comment (oliviachenairbnb) [reply pkg/backup/backupschedule/backup_schedule_manager.go]: updated @grovecai

File: pkg/backup/backupschedule/backup_schedule_manager.go (hunk 2 above)
Comment (oliviachenairbnb) [reply pkg/backup/backupschedule/backup_schedule_manager.go]: updated @grovecai

**Review by grovecai — APPROVED** [review]

**Comment by oliviachenairbnb** [comment]
seems my comment cannot trigger the ci. Can you please help run again? I just rebased the branch. Thanks @Ehco1996

**Comment by oliviachenairbnb** [comment]
also seeing we need two approval before merge from this doc https://github.com/pingcap/community/blob/master/contributors/README.md
Can you pls help refer another reviewer if that's needed? @grovecai Thanks.

**Comment by BornChanger** [comment]
Can you please add a case for your pr to cover the failing scenario?

**Pull request merged**